### PR TITLE
Chris becky psql

### PR DESCRIPTION
**Pull request by kozlowskicd**
Built route that creates a row in the goals table and an associated row in the progress table.  For tomorrow, need to build a user from the data provided by OAuth, currently, users can be created manually.

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -2,24 +2,41 @@
 
 require('dotenv').config();
 
+const pg = require('pg');
+const sql = require('./sql');
```
Comment (vladimirsan) [inline src/app.js:6]: what's the organization schema for these imports?

File: src/app.js [hunk 2]
```diff
@@ -29,43 +46,107 @@ passport.deserializeUser(function(obj, cb) {
   cb(null, obj);
 });
 
-const app = express();
-app.use(express.static('public'))
+app.use(express.static('public'));
 app.use(require('morgan')('combined'));
 app.use(require('body-parser').urlencoded({ extended: true }));
-app.use(require('express-session')({ secret: 'keyboard cat', resave: true, saveUninitialized: true }));
+app.use(require('express-session')({ secret: 'keyboard cat', resave: false, saveUninitialized: false }));
```
Comment (vladimirsan) [inline src/app.js:52]: let's move the secret to a .env

File: src/app.js [hunk 3]
```diff
@@ -29,43 +46,107 @@ passport.deserializeUser(function(obj, cb) {
...
+      userId: userData.id,
+      userName: userData.name,
+      userScreenName: userData.screen_name,
+      photoLink: request.user.photos[0].value,
+      oAuthToken: request.query.oauth_token,
+      oAuthVerifier: request.query.oauth_verifier,
+      oAuthTokenSecret: oAuthData.oauth_token_secret,
+    };
+
+    // add to database
```
Comment (vladimirsan) [inline src/app.js:81]: :) you know what I'm going to write right? no commented out messages!

File: src/app.js [hunk 4]
```diff
@@ -29,43 +46,107 @@ passport.deserializeUser(function(obj, cb) {
...
 });
 
+app.get('/test', (request, response) => {
+  return database.query(sql.test)
+    .then(result => {
+      if(result) {
+        response.send(result.rows);
+      }
+      else {
+        response.send('Whoops');
```
Comment (vladimirsan) [inline src/app.js:105]: Why 'Whoops' instead of a 401 or 400?

File: src/app.js [hunk 5]
```diff
@@ -29,43 +46,107 @@ passport.deserializeUser(function(obj, cb) {
...
+    .catch(console.error);
+});
+
+app.post('/createGoal', (request, response) => {
+
+  const paramsArray = [];
+
+  const paramsObject = request.body;
+
+  Object.keys(paramsObject).forEach(key => {
```
Comment (vladimirsan) [inline src/app.js:117]: Why do we need this code?

**Review by vladimirsan — CHANGES_REQUESTED** [review]

File: src/app.js (hunk 2 above)
Comment (kozlowskicd) [reply src/app.js:52]: the secret has been moved out

File: src/app.js (hunk 1 above)
Comment (kozlowskicd) [reply src/app.js:6]: We did a refactor today and reorganized our imports.  Third-party imports are above our module imports, and each group is in alphabetical order.

File: src/app.js (hunk 3 above)
Comment (kozlowskicd) [reply src/app.js:81]: removed!

File: src/app.js (hunk 4 above)
Comment (kozlowskicd) [reply src/app.js:105]: Was there for testing, but it's now changed to send 400 status.

File: src/app.js (hunk 5 above)
Comment (kozlowskicd) [reply src/app.js:117]: We need to create an array to pass the values through the SQL commands dynamically, and the data comes to us in an object.

**Pull request merged**